pkg/datagatherer/k8s/generic.go

```diff
@@ -0,0 +1,52 @@
+package k8s
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// The generic gatherer knows how to request a list of generic resources from
+// the Kubernetes apiserver.
+// It does not deserialize the objects into structured data, instead utilising
+// the Kubernetes `Unstructured` type for data handling.
+// This is to allow us to support arbitrary CRDs and resources that Preflight
+// does not have registered as part of its `runtime.Scheme`.
+type genericGatherer struct {
+	// The 'dynamic' client used for fetching data.
+	cl dynamic.Interface
+	// groupVersionResource is the name of the API group, version and resource
+	// that should be fetched by this data gatherer.
+	groupVersionResource schema.GroupVersionResource
+	// namespace, if specified, limits the namespace of the resources returned.
+	// This field *must* be omitted when the groupVersionResource refers to a
+	// non-namespaced resource.
+	namespace string
+}
+
+// Fetch will fetch the requested data from the apiserver, or return an error
+// if fetching the data fails.
+func (g *genericGatherer) Fetch() (interface{}, error) {
+	if g.groupVersionResource.Resource == "" {
+		return nil, fmt.Errorf("resource type must be specified")
+	}
+	resourceInterface := namespaceResourceInterface(g.cl.Resource(g.groupVersionResource), g.namespace)
+	list, err := resourceInterface.List(metav1.ListOptions{})
```

Comment:
We may want to consider allowing the ListOptions to be provided by the caller in future by adding a field to the `genericGatherer` struct. For now it isn't necessary though 😄

Reply:
Agree, it is an interesting feature for the future.